source/fundamentals/connection/mongoclientsettings.txt

```diff
@@ -0,0 +1,367 @@
+     - Sets the minimum amount of connections associated with a connection pool.
+
+Example
+~~~~~~~
+
+This example specifies the following driver behavior in a pool of
+``Connection`` types:
+
+- The thread to wait at most ``3 MINUTES`` for an available connection
+- To have at most ``200`` connections associated with the pool
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin ConnectionPoolSettings
+   :end-before: end ConnectionPoolSettings
+   :language: java
+   :emphasize-lines: 3-5
+   :dedent:
+
+.. _mcs-server-settings:
+
+Server Settings
+---------------
+
+Chain the `applyToServerSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToServerSettings(com.mongodb.Block)>`__
+method to modify the driver's behavior when monitoring each MongoDB
+server.
+
+The following table describes all the methods you can chain to your
+settings to modify the driver's behavior:
+
+.. list-table::
+   :widths: 40 60
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``addServerListener()``
+     - Adds a listener for server-related events.
+
+   * - ``addServerMonitorListener​()``
+     - Adds a listener for server monitor-related events.
+
+   * - ``applyConnectionString​()``
+     - Uses the settings from a ``ConnectionString`` object.
+
+   * - ``applySettings()``
+     - Uses the server settings specified in a ``ServerSettings`` object.
+
+   * - ``heartbeatFrequency()``
+     - Sets the interval for a cluster monitor to attempt reaching a server.
+
+   * - ``minHeartbeatFrequency()``
+     - Sets the interval for server monitoring checks.
+
+Example
+~~~~~~~
+
+This example specifies the following driver behavior in a MongoDB server:
+
+- To interval between server monitoring checks to be at least ``700 MILLISECONDS`` 
+- The cluster monitor to attempt reaching a server every ``15 SECONDS``
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin ServerSettings
+   :end-before: end ServerSettings
+   :language: java
+   :emphasize-lines: 3-5
+   :dedent:
+
+.. _mcs-ssl-settings:
+
+TLS/SSL Settings
+----------------
+
+Chain the `applyToSslSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToSslSettings(com.mongodb.Block)>`__
+method to modify the driver's behavior when using TLS/SSL to secure a
+connection between your application and MongoDB.
+
+The following table describes all the methods you can chain to your
+settings to modify the driver's behavior:
+
+.. list-table::
+   :widths: 40 60
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``applyConnectionString​()``
+     - Uses the settings from a ``ConnectionString`` object.
+
+   * - ``applySettings()``
+     - Uses the TLS/SSL settings specified in a ``SslSettings`` object.
+
+   * - ``context()``
+     -  Sets the secure socket protocol implementation to use when you enable TLS/SSL.
```

Comment:
It's not really the implementation, it's the context.  Just use what's in the Javadoc:

> Sets the SSLContext for use when TLS/SSL is enabled.